src/shared/calendarspec.c

```diff
@@ -1196,12 +1197,19 @@ static int tm_within_bounds(struct tm *tm, bool utc) {
                 return negative_errno();
 
         /* Did any normalization take place? If so, it was out of bounds before */
-        int cmp = CMP(t.tm_year, tm->tm_year) ?:
-                  CMP(t.tm_mon, tm->tm_mon) ?:
-                  CMP(t.tm_mday, tm->tm_mday) ?:
-                  CMP(t.tm_hour, tm->tm_hour) ?:
-                  CMP(t.tm_min, tm->tm_min) ?:
-                  CMP(t.tm_sec, tm->tm_sec);
+        if ((cmp = CMP(t.tm_year, tm->tm_year)) != 0)
+                t.tm_mon = 0;
+        else if ((cmp = CMP(t.tm_mon, tm->tm_mon)) != 0)
+                t.tm_mday = 1;
+        else if ((cmp = CMP(t.tm_mday, tm->tm_mday)) != 0)
+                t.tm_hour = 0;
+        else if ((cmp = CMP(t.tm_hour, tm->tm_hour)) != 0)
+                t.tm_min = 0;
+        else if ((cmp = CMP(t.tm_min, tm->tm_min)) != 0)
+                t.tm_sec = 0;
+        else if ((cmp = CMP(t.tm_sec, tm->tm_sec)) != 0) {
+                // no op
+        }
```

Comment:
drop the {} if there is no op to execute, i.e. end in:

```c
…
else if ((cmp = CMP(t.tm_min, tm->tm_min)) != 0)
        t.tm_sec = 0;
else
        cmp = CMP(t.tm_sec, tm->tm_sec);
```

Reply:
(also as per coding style https://systemd.io/CODING_STYLE/#formatting we use `//` style comments for FIXME stuff, and `/**/` comments for explanatory stuff. the no op comment is the latter... but I'd drop it entirely)

Reply:
Fixed.